src/chainstate/coordinator/comm.rs

```diff
@@ -0,0 +1,243 @@
+use std::{
+    thread, process
+};
+use std::time::{
+    Duration, Instant
+};
+use std::sync::{
+    Arc, RwLock,
+    atomic::{Ordering, AtomicU64}
+};
+
+use crossbeam_channel::{bounded, Sender, Receiver, TrySendError};
+
+
+/// Trait for use by the ChainsCoordinator
+/// 
+pub trait CoordinatorNotices {
+    fn notify_stacks_block_processed(&mut self);
+    fn notify_sortition_processed(&mut self);
+}
+
+pub struct ArcCounterCoordinatorNotices {
+    pub stacks_blocks_processed: Arc<AtomicU64>,
+    pub sortitions_processed: Arc<AtomicU64>
+}
+
+impl CoordinatorNotices for () {
+    fn notify_stacks_block_processed(&mut self) {}
+    fn notify_sortition_processed(&mut self) {}
+}
+
+impl CoordinatorNotices for ArcCounterCoordinatorNotices {
+    fn notify_stacks_block_processed(&mut self) {
+        self.stacks_blocks_processed.fetch_add(1, Ordering::SeqCst);
+    }
+    fn notify_sortition_processed(&mut self) {
+        self.sortitions_processed.fetch_add(1, Ordering::SeqCst);
+    }
+}
+
+/// Structure used for communication _with_ a running
+///   ChainsCoordinator
+#[derive(Clone)]
+pub struct CoordinatorChannels {
+    // ChainsCoordinator takes two kinds of signals:
+    //    new stacks block & new burn block
+    // These signals can be coalesced -- the coordinator doesn't need
```

Comment:
Comment typo?

Reply:
It might just be using odd terminology -- there's two communication channels the coordinator is listening on for notifications that it should wake up and try to process. One indicates that a burnchain block has been received and processed into the burnchain db, the other indicates the same for a staging stacks block.

Reply:
Updated comments in 2f94671